homeassistant/components/geo_location/nsw_rural_fire_service_feed.py

```diff
@@ -0,0 +1,275 @@
+                                  for entry in feed_entries}
+            # For entity management the external ids from the feed are used.
+            feed_external_ids = {entry.external_id
+                                 for entry in feed_entries}
+            remove_external_ids = self._managed_external_ids.difference(
+                feed_external_ids)
+            self._remove_entities(remove_external_ids)
+            update_external_ids = self._managed_external_ids.intersection(
+                feed_external_ids)
+            self._update_entities(update_external_ids)
+            create_external_ids = feed_external_ids.difference(
+                self._managed_external_ids)
+            self._generate_new_entities(create_external_ids)
+        elif status == geojson_client.UPDATE_OK_NO_DATA:
+            _LOGGER.debug("Update successful, but no data received from %s",
+                          self._feed)
+        else:
+            _LOGGER.warning("Update not successful, no data received from %s",
+                            self._feed)
+            # Remove all entities.
+            self._remove_entities(self._managed_external_ids.copy())
+
+    def _generate_new_entities(self, external_ids):
+        """Generate new entities for events."""
+        new_entities = []
+        for external_id in external_ids:
+            new_entity = NswRuralFireServiceLocationEvent(self, external_id)
+            _LOGGER.debug("New entity added %s", external_id)
+            new_entities.append(new_entity)
+            self._managed_external_ids.add(external_id)
+        # Add new entities to HA.
+        self._add_entities(new_entities, True)
+
+    def _update_entities(self, external_ids):
+        """Update entities."""
+        for external_id in external_ids:
+            _LOGGER.debug("Existing entity found %s", external_id)
+            dispatcher_send(self._hass, SIGNAL_UPDATE_ENTITY, external_id)
+
+    def _remove_entities(self, external_ids):
+        """Remove entities."""
+        for external_id in external_ids:
+            _LOGGER.debug("Entity not current anymore %s", external_id)
+            self._managed_external_ids.remove(external_id)
+            dispatcher_send(self._hass, SIGNAL_DELETE_ENTITY, external_id)
+
+    def get_feed_entry(self, external_id):
+        """Return a feed entry identified by external id."""
+        return self._feed_entries.get(external_id)
+
+
+class NswRuralFireServiceLocationEvent(GeoLocationEvent):
+    """This represents an external event with GeoJSON data."""
+
+    def __init__(self, feed_manager, external_id):
+        """Initialize entity with data from feed entry."""
+        self._feed_manager = feed_manager
+        self._external_id = external_id
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        async_dispatcher_connect(
+            self.hass, SIGNAL_DELETE_ENTITY, self._delete_callback)
```

Comment:
Make external_id part of the signal to avoid having to check the external_id in the callbacks. See eg:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mysensors/device.py#L108.

Reply:
Makes sense, thanks for the hint.